pkg/kp/healthmanager.go

```diff
@@ -0,0 +1,340 @@
+package kp
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/square/p2/Godeps/_workspace/src/github.com/Sirupsen/logrus"
+	"github.com/square/p2/Godeps/_workspace/src/github.com/hashicorp/consul/api"
+
+	"github.com/square/p2/pkg/logging"
+	"github.com/square/p2/pkg/util/stream"
+)
+
+type consulSimpleHealthManager struct {
+	store  consulStore
+	logger logging.Logger
+}
+
+type consulSimpleHealthUpdater struct {
+	store      consulStore
+	logger     logging.Logger
+	lastCheck  time.Time
+	lastStatus string
+}
+
+// newSimpleHealthManager creates a simple health manager that writes a service's health
+// status to Consul every 15 seconds or whenever its status changes.
+func (c consulStore) newSimpleHealthManager(
+	_node string,
+	logger logging.Logger,
+) HealthManager {
+	return &consulSimpleHealthManager{
+		store:  c,
+		logger: logger,
+	}
+}
+
+// NewUpdater implements the HealthManager interface.
+func (m *consulSimpleHealthManager) NewUpdater(_service string) HealthUpdater {
+	return &consulSimpleHealthUpdater{
+		store:  m.store,
+		logger: m.logger,
+	}
+}
+
+// Close implements the HealthManager interface. consulSimpleHealthManager has no
+// resources to release.
+func (m *consulSimpleHealthManager) Close() {}
+
+// PutHealth implements the HealthUpdater interface.
+func (u *consulSimpleHealthUpdater) PutHealth(health WatchResult) error {
+	if health.Status != u.lastStatus || time.Since(u.lastCheck) > TTL/4 {
```

Comment:
The appropriate advertisement should be: "it implements whatever behavior was there before, but this one uses a new interface."

And actually, I think a TTL of 60 seconds is far too long. 10-15 seconds seems more reasonable for detecting that a preparer has died and all the services on the machine could be in questionable state. (A 15s TTL would, incidentally, make the revised check equal to `oldTTL / 4`.)